### ISSUE TITLE
Parameters in from_data_frame() function

### ISSUE THREAD

**Issue by tlubitz** [tlubitz/SBtab#55]
@eladnoor 
A user had a question concerning your from_data_frame() function:

"[...]the function takes 6 arguments and 1 keyword argument, which seems rather excessive, considering all but the dataframe are used to generate the header exclusively; I think just passing the dataframe and a header would be preferable."

I didn't want to change your code but rather ask for your feedback on this.

**Comment by eladnoor** [comment]
I don't see a problem with that. The only issue is how the user will create the header object (or string). Maybe there should be another function for doing that.

**Comment by tlubitz** [comment]
The header object is not very restrictive in SBtab. It is obligatory to choose a TableType and of course the data frame should be there. Since we are not creating a SBtabDocument but "only" a SBtabTable in your function, it should suffice to have the parameters df and table_type, while the others can be optional. Wolf is also a friend of keeping these kinds of attributes numerous but mostly optional. Let me know what you think

**Comment by eladnoor** [comment]
Yes, that's even better. We can just make all other parameters optional and table_type obligatory

**Comment by tlubitz** [comment]
Cool! Will you implement it so I don't mess with your function? That would be great...

**Comment by eladnoor** [comment]
Okay, I did it and created a pull request.
However, I think there is a more fundamental issue that only you can solve. There is no easy way to create a new SBtab from within a script The constructor only works when you provide it with a full table string including the headers and the data rows. I think it should work more like a container class, so one should be able to easily create an empty SBtabTable (with mostly default parameters), then define the columns, and then add rows with data. After this is available, I will be happy to rewrite the from_data_frame() function to use these options instead of the current mess it is now.

**Comment by tlubitz** [comment]
Yes, this is a bigger issue. I will look into it and discuss with Wolf whether we make according changes. But although it is in not in the state of such a container class, I wouldn't call the current state a mess. It has worked well for quite some time; although it's surely not perfect, it is constantly improving. I'll update you if we do these changes, but it's not top priority right now. Thanks for the rewriting of the function, I'll look into it!

**Comment by eladnoor** [comment]
Sorry again for not being clear. I only meant that the function from_data_frame() that I wrote is messy, basically because I had to create a string containing the table data in the format that can be read by the constructor. It is far from ideal to copy raw data from one container to another using a string (slow, easy to break with any slight change to the syntax, truncation due to number formatting).
